cmd/juju/application/deployer/bundlehandler_test.go

```diff
@@ -107,8 +107,8 @@ func (s *BundleDeployRepositorySuite) TestDeployBundleSuccess(c *gc.C) {
 	s.expectEmptyModelToStart(c)
 	s.expectWatchAll()
 
-	mysqlCurl := charm.MustParseURL("ch:mysql")
-	wordpressCurl := charm.MustParseURL("ch:wordpress")
+	mysqlCurl := charm.MustParseURL("ch:xenial/mysql")
```

Comment:
This feels wrong and raises questions I need to find the answer to.

Reply:
I understand the gut feeling, I had the same. Let me try and explain why I think it's actually ok

These unit tests basically deploy a bundle from it's yaml from all the way to the CLI output. So firstly, charms are resolved and then applications are added, etc.

These charm urls are only used to check call args of `CharmInfo` and `Deploy`, both of which are called when adding applications

Since my change means that series are always (instead of sometimes) added to the charm url _whilst adding a charm_, by the time we deploy an application the charm urls being used always have a series

The gut feeling I had I think was because these charm urls feel like the input to something we're testing, which would mean the code  we're testing is broken if we provide a url without a series. But that's not the case. These urls are parsed mid way through the code we're testing, and here we are asserting that we have some specific charm urls

Reply:
If we look to the code how it is, we can already see that these are the series that the charm is added with, and the application deployed with:
https://github.com/juju/juju/blob/3.1/cmd/juju/application/deployer/bundlehandler_test.go#L140C3-L143

There are also some tests where adding the charm url is given a series, such as:
https://github.com/juju/juju/blob/3.1/cmd/juju/application/deployer/bundlehandler_test.go#L1546-L1572

Thoughts?

Reply:
the tests are far from what actually happens. :-( https://pastebin.canonical.com/p/xqR5yyK2Ws/